### PR TITLE
Update the code to use a non-deprecated header.

### PR DESCRIPTION
**Pull request by clalancette**
This requires a little additional work, since Humble does not have that header.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,15 +2,22 @@ cmake_minimum_required(VERSION 3.5)
...
 if(NOT CMAKE_CXX_STANDARD)
-  set(CMAKE_CXX_STANDARD 14)
+  set(CMAKE_CXX_STANDARD 17)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+message(STATUS "CHRIS: ROS_DISTRO: $ENV{ROS_DISTRO}")
```
Comment (tfoote) [inline CMakeLists.txt]: Debug statement left in ;-)

Comment (tfoote) [reply CMakeLists.txt]: Maybe a comment about when this clause can be removed in the future.

File: include/depthimage_to_laserscan/DepthImageToLaserScan.hpp [hunk 2]
```diff
@@ -38,7 +38,11 @@
 
 #include "depthimage_to_laserscan/DepthImageToLaserScan_export.h"
 #include "depthimage_to_laserscan/depth_traits.hpp"
+#ifdef LEGACY_PINHOLE_CAMERA_INCLUDE
```
Comment (tfoote) [inline include/depthimage_to_laserscan/DepthImageToLaserScan.hpp]: A comment here linking to the issue for future reference and removal trimming would be valuable.

File: CMakeLists.txt (hunk 1 above)
Comment (cottsay) [inline CMakeLists.txt]:
```suggestion
```

**Review by cottsay — COMMENTED** [review]
If you're going to bump to C++17 anyway, have you considered using `__has_include` instead?

https://en.cppreference.com/w/cpp/preprocessor/include

File: CMakeLists.txt (hunk 1 above)
Comment (clalancette) [reply CMakeLists.txt]: Excellent point.  I don't know how many times I've looked at this in the last 2 weeks, and I never saw it :).  Removed in 3eeba6c

**Comment by clalancette** [comment]
> If you're going to bump to C++17 anyway, have you considered using `__has_include` instead?

Oh, that's a good point.  Let me give that a try.

**Comment by clalancette** [comment]
> Oh, that's a good point. Let me give that a try.

Oh yeah, that's much nicer.  A much smaller patch that way.  Thanks!

File: include/depthimage_to_laserscan/DepthImageToLaserScan.hpp (hunk 2 above)
Comment (clalancette) [reply include/depthimage_to_laserscan/DepthImageToLaserScan.hpp]: Done in 3eeba6c

**Review by quarkytale — APPROVED** [review]

**Pull request merged**